hibernate-core/src/test/java/org/hibernate/test/hql/NullSelectExpressionTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.test.hql;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.junit.Test;
+
+import org.hibernate.Session;
+import org.hibernate.hql.internal.ast.QuerySyntaxException;
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * @author Gail Badner
+ */
+@TestForIssue( jiraKey = "HHH-10757")
+public class NullSelectExpressionTest extends BaseCoreFunctionalTestCase {
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		Object result = s.createQuery(
+				"select firstName, null, lastName from NullSelectExpressionTest$Person where lastName='Munster'"
+		).uniqueResult();
+		assertTrue( Object[].class.isInstance( result ) ) ;
+		Object[] objects = (Object[]) result;
+
+		assertEquals( 3, objects.length );
+		assertEquals( "Herman", objects[0] );
+		assertNull( objects[1] );
+		assertEquals( "Munster", objects[2] );
+
+		s.getTransaction().rollback();
+		s.close();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testAliasedNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
+		Object result = s.createQuery(
+				"select firstName, null as nullValue, lastName from NullSelectExpressionTest$Person where lastName='Munster'"
+		).uniqueResult();
+		assertTrue( Object[].class.isInstance( result ) ) ;
+		Object[] objects = (Object[]) result;
+		assertEquals( 3, objects.length );
+		assertEquals( "Herman", objects[0] );
+		assertNull( objects[1] );
+		assertEquals( "Munster", objects[2] );
+
+		s.getTransaction().rollback();
+		s.close();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HHH-10757")
+	public void testNewNullSelectExpression() {
+		Session s = openSession();
+		s.getTransaction().begin();
+		Person person = new Person();
+		person.firstName = "Herman";
+		person.middleName = "Joseph";
+		person.lastName = "Munster";
+		s.persist( person );
+		s.flush();
+		s.clear();
+
```

Comment:
The following fails because Hibernate assumes there are only 2 select expressions (id, firstName) instead of 4; there is no constructor with args:  long, java.lang.String.